actix-http/src/h1/decoder.rs

```diff
@@ -203,7 +203,14 @@ impl MessageType for Request {
 
                     (len, method, uri, version, req.headers.len())
                 }
-                httparse::Status::Partial => return Ok(None),
+                httparse::Status::Partial => {
+                    return if src.len() >= MAX_BUFFER_SIZE {
+                        trace!("MAX_BUFFER_SIZE unprocessed data reached, closing");
+                        Err(ParseError::TooLarge)
+                    } else {
+                        Ok(None)
+                    }
```

Comment:
Can't the `if` be moved directly into the match arm?

Reply:
you can but you need another match on the condition which would lead to another Pratian and return. So you end up more code doing the same thing with the same logic.

Reply:
Huh? I find 
```suggestion
                    httparse::Status::Partial if src.len() >= MAX_BUFFER_SIZE => {
                       error!("MAX_BUFFER_SIZE unprocessed data reached, closing");
                       Err(ParseError::TooLarge)
                    }
                    httparse::Status::Partial => Ok(None),
```

much clearer and simpler.

Reply:
It's also rather weird to use explicit `return` here why a in another arm `?` is used. I'd probably just create the `Result` and have one `?` at the end of the `match` block.

Reply:
... and probably add a note that `httparse::Status::Partial => return Ok(None)` short-circuits the `decode` function in case we don't have enough data yet. Easy to miss... 😅

Reply:
It really is a style choice. Your write more code for fewer lines and I don't find it's simple or clear as it's the same thing.

Reply:
Having all conditions directly on match arm allows to mentally filter all possible cases on the same level, also it allows to group match arms with the same outcome which is nice. But okay, I take your point about style choice...

Reply:
Since the condition has nothing to do with the variant matched I feel that its appropriate to use an inner if statement.